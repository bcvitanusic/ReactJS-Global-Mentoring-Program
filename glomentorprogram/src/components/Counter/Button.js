import React from 'react';

const el = React.createElement;

export function Button(props) {
	return el(
		'button',
		{ onClick: props.handleClick, className: 'counter-button' },
		props.name
	);
}
