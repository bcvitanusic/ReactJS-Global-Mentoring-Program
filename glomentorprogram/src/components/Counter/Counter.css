.container {
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
}

button {
    background-color: #424242; 
    text-align: center;
    padding: 5px 20px;
    border-radius: 10px;
    color: gold;
    font-size: 30px;
    border: none;	
    margin: 20px;
}

button:hover {
    cursor: pointer;
    background-color: #F65261;
    box-shadow: 0px 20px 30px gray;
    transition: all 0.35s;
}


h1 {
    color: #F65261;
    text-align: center;
}

p {
    
}