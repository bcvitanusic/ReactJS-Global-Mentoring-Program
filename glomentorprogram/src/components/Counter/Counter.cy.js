import Counter from './Counter';

describe('<Counter>', () => {
	it('checks if correct value displays as a prop', () => {
		cy.mount(<Counter num={5} />);
		cy.get('p').should('have.text', '5');
	});

	it('checks if increments value', () => {
		cy.mount(<Counter num={5} />);
		cy.get('button').contains('Increment').click();
		cy.get('p').should('have.text', '6');
	});

	it('checks if decrements value', () => {
		cy.mount(<Counter num={5} />);
		cy.get('button').contains('Decrement').click();
		cy.get('p').should('have.text', '4');
	});
});
