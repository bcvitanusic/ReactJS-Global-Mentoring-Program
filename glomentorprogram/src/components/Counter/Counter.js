import React from 'react';
import './Counter.css';

const el = React.createElement;

function Button(props) {
	return el(
		'button',
		{ onClick: props.handleClick, className: 'counter-button' },
		props.name
	);
}

class Counter extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			num: this.props.num,
		};
		this.handleIncrement = this.handleIncrement.bind(this);
		this.handleDecrement = this.handleDecrement.bind(this);
	}

	handleIncrement = () => {
		this.setState({
			num: this.state.num + 1,
		});
	};

	handleDecrement = () => {
		this.setState({
			num: this.state.num - 1,
		});
	};

	render() {
		return el('div', { className: 'container' }, [
			el('h1', null, 'COUNTER'),
			el('div', null, [
				el(
					Button,
					{
						handleClick: this.handleDecrement,
						name: 'Decrement',
					},
					null
				),
				el(
					Button,
					{ handleClick: this.handleIncrement, name: 'Increment' },
					null
				),
			]),

			el(
				'p',
				{
					className: 'paragraph',
				},
				null,
				this.state.num
			),
		]);
	}
}

export default Counter;
