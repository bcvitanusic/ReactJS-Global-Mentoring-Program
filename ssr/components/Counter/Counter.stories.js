import Counter from './Counter';

export default {
	title: 'Counter',
	component: Counter,
};

export const CounterComponent = () => <Counter num={5} />;
