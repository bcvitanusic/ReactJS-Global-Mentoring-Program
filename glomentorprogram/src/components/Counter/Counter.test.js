/* eslint-disable testing-library/await-async-query */
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Counter from './Counter';
import renderer from 'react-test-renderer';

describe('Counter', () => {
	const renderCounter = () => render(<Counter num={5} />);

	test('if props display correctly', () => {
		renderCounter();
		expect(screen.getByText(5)).toBeInTheDocument();
	});

	test('increments by click', () => {
		renderCounter();
		const buttonElement = screen.getByText(/increment/i);
		expect(buttonElement).toBeInTheDocument();
		userEvent.click(buttonElement);
		expect(screen.getByText(6)).toBeInTheDocument();
	});

	test('decrements by click', () => {
		renderCounter();
		const buttonElement = screen.getByText('Decrement');
		expect(buttonElement).toBeInTheDocument();
		userEvent.click(buttonElement);
		expect(screen.getByText(4)).toBeInTheDocument();
	});
});

describe('Counter snapshot', () => {
	it('renders', () => {
		const component = renderer.create(<Counter num={5} />);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('decrements value', () => {
		const component = renderer.create(<Counter num={5} />);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();

		component.root.findAllByType('button')[0].props.onClick();
		tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('increments value', () => {
		const component = renderer.create(<Counter num={5} />);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();

		component.root.findAllByType('button')[1].props.onClick();
		tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
