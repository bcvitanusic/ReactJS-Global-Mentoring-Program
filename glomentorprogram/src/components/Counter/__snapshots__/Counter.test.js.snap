// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter snapshot decrements value 1`] = `
<div
  className="container"
>
  <h1>
    COUNTER
  </h1>
  <div>
    <button
      className="counter-button"
      onClick={[Function]}
    >
      Decrement
    </button>
    <button
      className="counter-button"
      onClick={[Function]}
    >
      Increment
    </button>
  </div>
  <p
    className="paragraph"
  >
    5
  </p>
</div>
`;

exports[`Counter snapshot decrements value 2`] = `
<div
  className="container"
>
  <h1>
    COUNTER
  </h1>
  <div>
    <button
      className="counter-button"
      onClick={[Function]}
    >
      Decrement
    </button>
    <button
      className="counter-button"
      onClick={[Function]}
    >
      Increment
    </button>
  </div>
  <p
    className="paragraph"
  >
    4
  </p>
</div>
`;

exports[`Counter snapshot increments value 1`] = `
<div
  className="container"
>
  <h1>
    COUNTER
  </h1>
  <div>
    <button
      className="counter-button"
      onClick={[Function]}
    >
      Decrement
    </button>
    <button
      className="counter-button"
      onClick={[Function]}
    >
      Increment
    </button>
  </div>
  <p
    className="paragraph"
  >
    5
  </p>
</div>
`;

exports[`Counter snapshot increments value 2`] = `
<div
  className="container"
>
  <h1>
    COUNTER
  </h1>
  <div>
    <button
      className="counter-button"
      onClick={[Function]}
    >
      Decrement
    </button>
    <button
      className="counter-button"
      onClick={[Function]}
    >
      Increment
    </button>
  </div>
  <p
    className="paragraph"
  >
    6
  </p>
</div>
`;

exports[`Counter snapshot renders 1`] = `
<div
  className="container"
>
  <h1>
    COUNTER
  </h1>
  <div>
    <button
      className="counter-button"
      onClick={[Function]}
    >
      Decrement
    </button>
    <button
      className="counter-button"
      onClick={[Function]}
    >
      Increment
    </button>
  </div>
  <p
    className="paragraph"
  >
    5
  </p>
</div>
`;
